job/server/src/main/java/alluxio/worker/job/task/TaskExecutor.java

```diff
@@ -47,50 +49,66 @@
       RunTaskContext context, TaskExecutorManager taskExecutorManager) {
     mJobId = jobId;
     mTaskId = taskId;
-    mJobConfig = jobConfig;
-    mTaskArgs = taskArgs;
+    mRunTaskCommand = runTaskCommand;
     mContext = Preconditions.checkNotNull(context);
     mTaskExecutorManager = Preconditions.checkNotNull(taskExecutorManager);
   }
 
   @Override
   public void run() {
-    // TODO(yupeng) set other logger
+    JobConfig jobConfig = null;
+    Serializable taskArgs = null;
+    try {
+      jobConfig = (JobConfig) SerializationUtils.deserialize(
+          mRunTaskCommand.getJobConfig().toByteArray());
+      if (mRunTaskCommand.hasTaskArgs()) {
+        taskArgs = SerializationUtils.deserialize(mRunTaskCommand.getTaskArgs().toByteArray());
+      }
+    } catch (IOException | ClassNotFoundException e) {
+      fail(e, jobConfig, null);
+    }
+
     PlanDefinition<JobConfig, Serializable, Serializable> definition;
     try {
-      definition = PlanDefinitionRegistry.INSTANCE.getJobDefinition(mJobConfig);
+      definition = PlanDefinitionRegistry.INSTANCE.getJobDefinition(jobConfig);
     } catch (JobDoesNotExistException e) {
-      LOG.error("The job definition for config {} does not exist.", mJobConfig.getName());
+      LOG.error("The job definition for config {} does not exist.", jobConfig.getName());
+      fail(e, jobConfig, taskArgs);
       return;
     }
 
     mTaskExecutorManager.notifyTaskRunning(mJobId, mTaskId);
     Serializable result;
     try {
-      result = definition.runTask(mJobConfig, mTaskArgs, mContext);
+      result = definition.runTask(jobConfig, taskArgs, mContext);
     } catch (InterruptedException e) {
       // Cleanup around the interruption should already have been handled by a different thread
       Thread.currentThread().interrupt();
       return;
     } catch (Throwable t) {
-      if (ServerConfiguration.getBoolean(PropertyKey.DEBUG)) {
-        mTaskExecutorManager.notifyTaskFailure(mJobId, mTaskId, ExceptionUtils.getStackTrace(t));
-      } else {
-        mTaskExecutorManager.notifyTaskFailure(mJobId, mTaskId, t.getMessage());
-      }
-      LOG.warn("Exception running task for job {}({}) : {}", mJobConfig.getName(),
-          mTaskArgs.toString(), t.getMessage());
+      fail(t, jobConfig, taskArgs);
       return;
     }
     mTaskExecutorManager.notifyTaskCompletion(mJobId, mTaskId, result);
   }
+
+  private void fail(Throwable t, JobConfig jobConfig, Serializable taskArgs) {
+    if (ServerConfiguration.getBoolean(PropertyKey.DEBUG)) {
+      mTaskExecutorManager.notifyTaskFailure(mJobId, mTaskId, ExceptionUtils.getStackTrace(t));
+    } else {
+      mTaskExecutorManager.notifyTaskFailure(mJobId, mTaskId, t.getMessage());
+    }
+    LOG.warn("Exception running task for job {}({}) : {}",
+        (jobConfig == null) ? "" : jobConfig.getName(),
+        (taskArgs == null) ? "" : taskArgs.toString(), t.getMessage());
```

Comment:
it may be more helpful to have special values for null instead of empty string when reading the logs.

Reply:
Addressed.